modules/services/himitsu.nix

```diff
@@ -0,0 +1,55 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let cfg = config.services.himitsu;
+in {
+  options = {
+    services.himitsu = {
+      enable = mkEnableOption ''
+        Himitsu, secret storage system for Unix systems
+      '';
+
+      package = mkOption {
+        type = types.package;
+        default = pkgs.himitsu;
+        defaultText = literalExpression "pkgs.himitsu";
+        description = "The himitsu package to use";
+      };
+
+      prompterPackage = mkOption {
+        type = types.package;
+        default = pkgs.hiprompt-gtk-py;
+        defaultText = literalExpression "pkgs.hiprompt-gtk-py";
+        description = "The himitsu prompter package to use";
+      };
+
+      prompter = mkOption {
+        type = types.str;
+        default = "${cfg.prompterPackage}/bin/hiprompt-gtk";
+        defaultText =
+          literalExpression "\${cfg.prompterPackage}/bin/hiprompt-gtk";
+        description = "The himitsu prompter command to use";
+      };
```

Comment:
Rather than have `prompterPackage` and `prompter`, just have `prompter = mkPackageOption "himitsu prompter" [ default = [ "hiprompt-gtk-py" ]; };` and use `lib.getExe` to generate the configuration file.

Reply:
we have to use `lib.getExe'` and specify the name of the executable explicitly (unless `meta.mainProgram` is specified accordingly by the corresponding package.